node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -176,6 +180,9 @@ case class DataMessageHandler(
         Future.successful(this)
       case HeadersMessage(count, headers) =>
         logger.info(s"Received headers message with ${count.toInt} headers")
+        if (count.toInt == 0) {
+          Try(initialSyncDone.map(_.success(Done)))
+        }
```

Comment:
If we're doing IBD and finish downloading block headers this will signal that we are done syncing even though we have no filter headers or filters